fsm/config.go

```diff
@@ -0,0 +1,118 @@
+package fsm
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+// Config contains the states, commands, functions and
+// default messages of the FSM
+type Config struct {
+	States    []string   `yaml:"states"`
+	Commands  []string   `yaml:"commands"`
+	Functions []Function `yaml:"functions"`
+	Defaults  Defaults   `yaml:"defaults"`
+}
+
+// Function lists the transitions available for the FSM
+type Function struct {
+	Transition Transition `yaml:"transition"`
+	Command    string     `yaml:"command"`
+	Slot       Slot       `yaml:"slot"`
+	Extension  string     `yaml:"extension"`
```

Comment:
Instead of specifying `ext_foo` in the message I added a dedicated `Extension` field. Where one would choose to use an extension instead of message.

Reply:
Was thinking about making this change eventually, thank you for anticipating it!